src/wp-includes/blocks.php

```diff
@@ -760,6 +760,29 @@ function get_hooked_blocks( $name ) {
 	return $hooked_blocks;
 }
 
+function insert_hooked_blocks( $block, $parent = null, $block_index = null, $chunk_index = null ) {
+	$hooked_blocks = get_hooked_blocks( $block['blockName'] );
+	foreach ( $hooked_blocks as $hooked_block_type => $relative_position ) {
+		$hooked_block = array(
+			'blockName'    => $hooked_block_type,
+			'attrs'        => array(),
+			'innerHTML'    => '',
+			'innerContent' => array(),
+			'innerBlocks'  => array(),
+		);
+		if ( 'before' === $relative_position ) {
+			insert_inner_block( $parent, $block_index, $chunk_index, $hooked_block );
+		} elseif ( 'after' === $relative_position ) {
+			insert_inner_block( $parent, $block_index + 1, $chunk_index, $hooked_block );
+		} elseif ( 'first_child' === $relative_position ) {
+			prepend_inner_block( $block, $hooked_block );
```

Comment:
One observation. In this scenario, I assume that `$parent` is the same as `$block`. The code would be easier to follow if that was explicitly expresses as `$parent`. 

By the way, for consistency, the variable could become `$parent_block`.

Reply:
> One observation. In this scenario, I assume that `$parent` is the same as `$block`.

They're different, see https://github.com/WordPress/wordpress-develop/pull/5247/files#diff-8c99af92e4ec0fdb307ddd9b42be1e1ef1efe4a9f31287c23f346244dddd1ce9R763.

`$block` is the block that tree traversal is currently processing, `$parent` is its parent. First child/last child insertion thus needs to happen to `$block` (whereas sibling insertion needs to happen to `$parent`'s `innerBlocks`; however, note my [other comment](https://github.com/WordPress/wordpress-develop/pull/5247/files#r1331151853)).